src/cts/src/TritonCTS.cpp

```diff
@@ -1333,4 +1340,190 @@ double TritonCTS::computeInsertionDelay(const std::string& name,
   return insDelayPerMicron;
 }
 
+void addDummyCell(ClockInst*& inst,
+                  Clock::SubNet& subNet,
+                  const std::vector<sta::LibertyCell*>& dummyCandidates)
+{
+}
+
+void TritonCTS::writeDummyLoadsToDb(Clock& clockNet)
+{
+  // Traverse clock tree and compute ideal output caps for clock
+  // buffers in the same level
+  if (!computeIdealOutputCaps(clockNet)) {
+    // No cap adjustment is needed
+    return;
+  }
+
+  // Find suitable candidate cells for dummy loads
+  std::vector<sta::LibertyCell*> dummyCandidates;
+  findCandidateDummyCells(dummyCandidates);
+
+  clockNet.forEachSubNet([&](Clock::SubNet& subNet) {
+    subNet.forEachSink([&](ClockInst* inst) {
+      if (inst->isClockBuffer()
+          && !sta::fuzzyEqual(inst->getOutputCap(),
+                              inst->getIdealOutputCap())) {
+        addDummyCell(inst, subNet, dummyCandidates);
+      }
+    });
+  });
+}
+
+// Return true if any clock buffers need cap adjustment; false otherwise
+bool TritonCTS::computeIdealOutputCaps(Clock& clockNet)
+{
+  bool needAdjust = false;
+
+  // pass 1: compute actual output caps seen by each clock instance
+  clockNet.forEachSubNet([&](const Clock::SubNet& subNet) {
+    ClockInst* driver = subNet.getDriver();
+    float sinkCapTotal = 0.0;
+    subNet.forEachSink([&](ClockInst* inst) {
+      odb::dbITerm* inputPin = inst->isClockBuffer()
+                                   ? getFirstInput(inst->getDbInst())
+                                   : inst->getDbInputPin();
+      float cap = getInputPinCap(inputPin);
+      // TODO: include wire caps?
+      sinkCapTotal += cap;
+    });
+    driver->setOutputCap(sinkCapTotal);
+  });
+
+  // pass 2: compute ideal output caps for perfectly balanced tree
+  clockNet.forEachSubNet([&](const Clock::SubNet& subNet) {
+    ClockInst* driver = subNet.getDriver();
+    float maxCap = std::numeric_limits<float>::min();
+    subNet.forEachSink([&](ClockInst* inst) {
+      if (inst->isClockBuffer() && inst->getOutputCap() > maxCap) {
```

Comment:
Should we be using the input cap for a sink?

Reply:
No, this is to match loads seen at output pin pins.